compiler/ilgen/TypeDictionary.cpp

```diff
@@ -577,6 +586,47 @@ TypeDictionary::CloseStruct(const char *structName)
    }
 
 TR::IlType *
+TypeDictionary::DefineUnion(const char *unionName)
+   {
+   TR_HashId unionID=0;
+   _unionsByName->locate(unionName, unionID);
+
+   UnionType *newType = new (PERSISTENT_NEW) UnionType(unionName, _trMemory);
+   _unionsByName->add(unionName, unionID, (void *) newType);
+
+   return (TR::IlType *) newType;
+   }
+
+void
+TypeDictionary::UnionField(const char *unionName, const char *fieldName, TR::IlType *type)
+   {
+   TR_HashId unionID=0;
+   if (_unionsByName->locate(unionName, unionID))
+      {
+      UnionType *unionType = (UnionType *) _unionsByName->getData(unionID);
+      unionType->AddField(fieldName, type);
```

Comment:
Okay yes, it does fall in the general category of making JitBuilder more resilient and documenting under #500 is fine.